url7507f7ef16_Neuro17_jsongkick_tgz-pJ8-jsongkick_AppJ8/src/jsongkick/App.java

```diff
@@ -0,0 +1,128 @@
+package jsongkick;
+
+import java.net.URISyntaxException;
+import java.util.ArrayList;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import search.ArtistSearch;
+import search.EventSearch;
+import search.LocationSearch;
+
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+
+import entity.Artist;
+import entity.Concert;
+import entity.FullLocation;
+
+
+public class App {
+	
+	private static final Logger log = LogManager.getLogger(App.class);
+
+	public static void run() throws URISyntaxException{
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 14. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test.

We recommend that you simplify this method or break it into multiple methods. For example, consider extracting the code block on lines 39-118 into a separate method.